nalgebra-sparse/src/cs.rs

```diff
@@ -543,3 +544,147 @@ pub fn convert_counts_to_offsets(counts: &mut [usize]) {
+
+            // Test that major offsets are monotonically increasing
+            if range_start > range_end {
+                return Err(SparseFormatError::from_kind_and_msg(
+                    SparseFormatErrorKind::InvalidStructure,
+                    "Offsets are not monotonically increasing.",
+                ));
+            }
+
+            let minor_idx_in_lane = minor_indices.get(range_start..range_end).ok_or(
+                SparseFormatError::from_kind_and_msg(
+                    SparseFormatErrorKind::IndexOutOfBounds,
+                    "A major offset is out of bounds.",
+                ),
+            )?;
+
+            // We test for in-bounds, uniqueness and monotonicity at the same time
+            // to ensure that we only visit each minor index once
+            let mut prev = None;
+            let mut nonmonotonic = false;
+
+            for &minor_idx in minor_idx_in_lane {
+                if minor_idx >= minor_dim {
+                    return Err(SparseFormatError::from_kind_and_msg(
+                        SparseFormatErrorKind::IndexOutOfBounds,
+                        "A minor index is out of bounds.",
+                    ));
+                }
+
+                if let Some(prev) = prev {
+                    if prev == minor_idx {
+                        return Err(SparseFormatError::from_kind_and_msg(
+                            SparseFormatErrorKind::DuplicateEntry,
+                            "Input data contains duplicate entries.",
+                        ));
+                    }
+                    if prev > minor_idx {
+                        if !sort {
+                            panic!(
+                                "Minor indices are not monotonically increasing within each lane while sorting is not expected." 
+                            );
+                        }
+                        nonmonotonic = true;
+                    }
+                }
+                prev = Some(minor_idx);
+            }
+
+            // sort if indices are nonmonotonic and sorting is expected
+            if nonmonotonic && sort {
+                let range_size = range_end - range_start;
+                minor_index_permutation.resize(range_size, 0);
+                compute_sort_permutation(&mut minor_index_permutation, &minor_idx_in_lane);
+                minor_idx_buffer.clear();
+                minor_idx_buffer.extend_from_slice(&minor_idx_in_lane);
+                apply_permutation(
+                    &mut minor_indices[range_start..range_end],
+                    &minor_idx_buffer,
+                    &minor_index_permutation,
+                );
+
+                // sort values if they exist
+                if let Some(values) = values_option.as_mut() {
+                    if values.len() == 0 {
+                        panic!(
+                            "Internal error: Sorting currently not supported if values are empty."
+                        );
```

Comment:
I don't immediately see any reason for this `panic`. Can it be directly removed?

Reply:
Yes, I also think, it would be better to remove it.